-1

**Failed Tests**:   Build 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-34c498/15140/summary.html
**COMMIT**: c290cb4292eabd8c68f3eb650becfa805ddab473
**CMSSW**: CMSSW_12_0_X_2021-05-17-1100/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/33755/15140/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - cms-sw/cmssw#33711 @silviodonato: Remove hardcoded number (255) about the L1 prefiring
 - cms-sw/cmssw#33717 @hatakeyamak: Cleaning up PF contents in AOD.
 - cms-sw/cmssw#33680 @fabiocos: MTD reconstruction: code cleaning for old comments (issue #24452)
 - cms-sw/cmssw#33746 @civanch: Added a new run time protection for Full Sim

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-34c498/15140/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-34c498/15140/git-merge-result


## Build

I found compilation error when building: 

<pre>/cvmfs/cms-ib.cern.ch/nweek-02681/slc7_amd64_gcc900/external/gcc/9.3.0/bin/../lib/gcc/x86_64-unknown-linux-gnu/9.3.0/../../../../x86_64-unknown-linux-gnu/bin/ld: tmp/slc7_amd64_gcc900/src/CondCore/ESSources/plugins/CondCoreESSourcesPlugins/CondDBESSource.cc.o: in function `cond::persistency::convertoToOracleConnection(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
CondDBESSource.cc:(.text._ZN4cond11persistency26convertoToOracleConnectionERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN4cond11persistency26convertoToOracleConnectionERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x8f0): undefined reference to `boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::do_assign(char const*, char const*, unsigned int)'
/cvmfs/cms-ib.cern.ch/nweek-02681/slc7_amd64_gcc900/external/gcc/9.3.0/bin/../lib/gcc/x86_64-unknown-linux-gnu/9.3.0/../../../../x86_64-unknown-linux-gnu/bin/ld: CondDBESSource.cc:(.text._ZN4cond11persistency26convertoToOracleConnectionERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN4cond11persistency26convertoToOracleConnectionERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x940): undefined reference to `boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::do_assign(char const*, char const*, unsigned int)'
/cvmfs/cms-ib.cern.ch/nweek-02681/slc7_amd64_gcc900/external/gcc/9.3.0/bin/../lib/gcc/x86_64-unknown-linux-gnu/9.3.0/../../../../x86_64-unknown-linux-gnu/bin/ld: CondDBESSource.cc:(.text._ZN4cond11persistency26convertoToOracleConnectionERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN4cond11persistency26convertoToOracleConnectionERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x996): undefined reference to `boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::do_assign(char const*, char const*, unsigned int)'
/cvmfs/cms-ib.cern.ch/nweek-02681/slc7_amd64_gcc900/external/gcc/9.3.0/bin/../lib/gcc/x86_64-unknown-linux-gnu/9.3.0/../../../../x86_64-unknown-linux-gnu/bin/ld: CondDBESSource.cc:(.text._ZN4cond11persistency26convertoToOracleConnectionERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN4cond11persistency26convertoToOracleConnectionERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0xac2): undefined reference to `boost::re_detail_107500::perl_matcher<__gnu_cxx::__normal_iterator<char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<boost::sub_match<__gnu_cxx::__normal_iterator<char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::construct_init(boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > > const&, boost::regex_constants::_match_flags)'
collect2: error: ld returned 1 exit status
gmake: *** [tmp/slc7_amd64_gcc900/src/CondCore/ESSources/plugins/CondCoreESSourcesPlugins/libCondCoreESSourcesPlugins.so] Error 1
Leaving library rule at src/CondCore/ESSources/plugins
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-05-17-1100/src/CondCore/ESSources/test/TestCondCoreESSources.cpp
>> Building binary UnitTestConcurrentIOVsCondCore
Copying tmp/slc7_amd64_gcc900/src/CondCore/ESSources/test/UnitTestConcurrentIOVsCondCore/UnitTestConcurrentIOVsCondCore to productstore area:
</pre>

--- a/CalibTracker/SiStripLorentzAngle/BuildFile.xml
+++ b/CalibTracker/SiStripLorentzAngle/BuildFile.xml
@@ -5,6 +5,7 @@
 <use name="DQMServices/Core"/>
 <use name="MagneticField/Records"/>
 <use name="root"/>
+<use name="boost_regex"/>
 <export>
   <lib name="1"/>
 </export>

--- a/CommonTools/Utils/BuildFile.xml
+++ b/CommonTools/Utils/BuildFile.xml
@@ -6,6 +6,7 @@
 <use name="FWCore/Utilities"/>
 <use name="FWCore/MessageLogger"/>
 <use name="boost"/>
+<use name="boost_regex"/>
 <use name="roothistmatrix"/>
 <export>
   <lib name="1"/>

--- a/CondCore/ESSources/plugins/BuildFile.xml
+++ b/CondCore/ESSources/plugins/BuildFile.xml
@@ -1,6 +1,7 @@
 <use name="FWCore/Catalog"/>
 <use name="FWCore/Framework"/>
 <use name="CondCore/ESSources"/>
+<use name="boost_regex"/>
 <library file="*.cc" name="CondCoreESSourcesPlugins">
   <flags EDM_PLUGIN="1"/>
 </library>

--- a/DQMServices/StreamerIO/plugins/BuildFile.xml
+++ b/DQMServices/StreamerIO/plugins/BuildFile.xml
@@ -5,6 +5,7 @@
 <use name="DQMServices/Core"/>
 <use name="DQMServices/Components"/>
 <use name="stdcxx-fs"/>
+<use name="boost_regex"/>
 <library file="*.cc" name="DQMServicesStreamerIOPlugins">
   <flags EDM_PLUGIN="1"/>
   <flags REM_CXXFLAGS="-Werror=unused-variable"/>

--- a/DataFormats/PatCandidates/BuildFile.xml
+++ b/DataFormats/PatCandidates/BuildFile.xml
@@ -31,6 +31,7 @@
 <use name="FWCore/ParameterSet"/>
 <use name="FWCore/Reflection"/>
 <use name="boost"/>
+<use name="boost_regex"/>
 <flags LCG_DICT_HEADER="classes_objects.h classes_trigger.h classes_user.h classes_other.h"/>
 <flags LCG_DICT_XML="classes_def_objects.xml classes_def_trigger.xml classes_def_user.xml classes_def_other.xml"/>
 <export>

--- a/EventFilter/L1TRawToDigi/BuildFile.xml
+++ b/EventFilter/L1TRawToDigi/BuildFile.xml
@@ -5,6 +5,7 @@
 <use name="DataFormats/L1DTTrackFinder"/>
 <use name="DataFormats/L1TMuon"/>
 <use name="EventFilter/RPCRawToDigi"/>
+<use name="boost_regex"/>
 <export>
   <lib name="1"/>
 </export>

--- a/HLTrigger/Timer/plugins/BuildFile.xml
+++ b/HLTrigger/Timer/plugins/BuildFile.xml
@@ -9,5 +9,6 @@
   <use name="FWCore/ServiceRegistry"/>
   <use name="FWCore/Utilities"/>
   <use name="HLTrigger/Timer"/>
+  <use name="boost_regex"/>
   <flags EDM_PLUGIN="1"/>
 </library>

--- a/RecoLocalTracker/SiPixelRecHits/BuildFile.xml
+++ b/RecoLocalTracker/SiPixelRecHits/BuildFile.xml
@@ -1,4 +1,5 @@
 <use name="boost"/>
+<use name="boost_regex"/>
 <use name="cuda"/>
 <use name="vdt_headers"/>
 <use name="CalibTracker/SiPixelESProducers"/>

--- a/RecoLuminosity/LumiProducer/plugins/BuildFile.xml
+++ b/RecoLuminosity/LumiProducer/plugins/BuildFile.xml
@@ -51,6 +51,7 @@
 <library file="CMSRunSummary2DB.cc" name="CMSRunSummary2DB">
   <use name="RecoLuminosity/LumiProducer"/>
   <use name="rootcore"/>
+  <use name="boost_regex"/>
   <flags EDM_PLUGIN="1"/>
 </library>
 

--- a/RecoLuminosity/LumiProducer/test/BuildFile.xml
+++ b/RecoLuminosity/LumiProducer/test/BuildFile.xml
@@ -44,11 +44,13 @@
   <use name="CoralBase"/>
   <use name="RelationalAccess"/>
   <use name="boost"/>
+  <use name="boost_regex"/>
 </bin>
 
 <bin file="testBoostRegex.cpp" name="testBoostRegex">
   <flags NO_TESTRUN="1"/>
   <use name="boost"/>
+  <use name="boost_regex"/>
 </bin>
 
 <bin file="validation/populateDummy2LumiDB.cpp" name="populateDummy2LumiDB">

